UnitySamples/FBLeaderboard/Docs/FBLeaderboardTutorial.md

```diff
@@ -0,0 +1,301 @@
+
+```c#
+string fbAccessToken = Facebook.Unity.AccessToken.CurrentAccessToken.TokenString;
+m_chilliConnect.PlayerAccounts.LogInUsingFacebook(fbAccessToken, (request, response) => OnChilliConnectFBLoggedIn(response), (request, error) => OnChilliConnectLoginFailed(error));
+```
+
+A handy feature is that the response contains the player's FB name:
+
+```c#
+private void OnChilliConnectFBLoggedIn(LogInUsingFacebookResponse response)
+{
+    Debug.Log("ChilliConnect logged in with FB account for " + response.FacebookName);
+}
+```
+
+### Linking FB to a ChilliConnect account
+
+When 'LogInUsingFacebook' is called (either after FB init or via user button press), ChilliConnect will attempt to find a player account associated with the given FB access token. If no account exists then the FB player is not yet associated with a ChilliConnect account and we link the current logged in FB account with the current local ChilliConnect account. Future attempts to login with the FB access token will return this account.
+
+```c#
+private void OnChilliConnectLoginFailed(LogInUsingFacebookError error)
+{
+    if(error.ErrorCode == LogInUsingFacebookError.Error.LoginNotFound)
+    {
+        LinkFacebookAccountRequestDesc requestDesc = new LinkFacebookAccountRequestDesc(m_fbAccessToken);
+        m_chilliConnect.PlayerAccounts.LinkFacebookAccount(requestDesc, (request, linkResponse) => OnChilliConnectLinked(linkResponse), (request, linkError) => Debug.LogError(linkError.ErrorDescription));
+    }
+}
+```
+
+### Logging into FB
+
+If the player has yet to login to Facebook then they may do so by pressing the FB button that performs the following function:
+
+```c#
+public void Login()
+{
+    FB.LogInWithReadPermissions(new string[] {"public_profile", "user_friends"}, OnFBLoginComplete);
+}
+```
+
+Facebook allows you to login with either read permissions or publish permissions. In this demo we only need the above read permissions in order to access the name and profile image of the player and their friends.
+
+Calling login causes one of two things to happen:
+
+1. If no session exists the FB login dialogue is shown to the player asking them to login.
+2. If the player has already granted your app permission and the session is still valid no dialogue will show.
+
+Successful login will return an access token and a similar flow as above is performed where we attempt to login to ChilliConnect with the access token and link FB to the local account if no other account is found.
+
+#### Existing accounts
+
+It may be the case (post re-install or on a different device for example) that the player logs in to ChilliConnect with a FB token and already has an account. The player may have even made progress in the game forgetting that they had an existing account. Different games handle this in different ways but the most common solution is to determine whether the local anonymous account or the FB linked account has more progress and use that data.
+
```

Comment:
Want to add anything about the approach used in this example? (in the code it says the FB account will switch to the current user).